From 2d022ee77d352d6fc479b883f62382e2cd314d90 Mon Sep 17 00:00:00 2001
From: Ryan McCabe <rmccabe@redhat.com>
Date: Tue, 3 Mar 2015 14:26:38 -0500
Subject: [PATCH] rgmanager: Fix potential NULL ptr deref in clustat

Fix a potential null pointer dereference that can very rarely
happen in clustat when running with -i.

Resolves: rhbz#1079207

Signed-off-by: Ryan McCabe <rmccabe@redhat.com>
---
 rgmanager/src/utils/clustat.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/rgmanager/src/utils/clustat.c b/rgmanager/src/utils/clustat.c
index d7e8dd8..56e2a15 100644
--- a/rgmanager/src/utils/clustat.c
+++ b/rgmanager/src/utils/clustat.c
@@ -79,6 +79,9 @@ flag_rgmanager_nodes(cluster_member_list_t *cml)
 
 	struct timeval tv;
 
+	if (!cml)
+		return;
+
 	if (msg_open(MSG_SOCKET, 0, 0, &ctx, 10) < 0) {
 		//perror("msg_open");
 		return;
-- 
2.1.0

